File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> hm = new HashMap<>();
        for (String p : participant) {
            if (hm.containsKey(p)) {
                hm.put(p, hm.get(p) + 1);
            }else{
                hm.put(p, 1);
            }
        }
        
        for (String c : completion) {
            hm.put(c, hm.get(c) - 1);
            if (hm.get(c) == 0) hm.remove(c);
        }
        
        for (String key : hm.keySet()) {
            answer = key;
        }
        
        
        
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <bits/stdc++.h>

using namespace std;

map<string, int> m;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";

    for (string p : participant) {
        m[p]++;
    }

    for (string c : completion) {
        m[c]--;
    }

    for (string p : participant) {
        if (m[p]) {
            answer = p;
            break;
        }
    }

    return answer;
}